nodebuilder/header/service.go

```diff
@@ -51,6 +52,22 @@ func (s *Service) GetVerifiedRangeByHeight(
 }
 
 func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	head, err := s.syncer.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
For now, we should also check whether we synced up to the requested height through LocalHead/StoreHead and if not, return an error

Reply:
We should also have this logic with TODO to remove this after https://github.com/celestiaorg/go-header/issues/32 is fixed

Reply:
> LocalHead/StoreHead

I'm checking it at line 68

Reply:
No. Syncer head is network head, while localhead can be different

Reply:
Ah, you mean the HasAt. Lines shifted after recent changes...

Reply:
We can't get `head.Height`, without getting the Head. So it should be reworked to `Head` request. I wanted to avoid this...., but ok

Reply:
604757d90fad4106753fabd917c9fcae169bcfd2

Reply:
> We can't get head.Height, without getting the Head. So it should be reworked to Head request. I wanted to avoid this...., but ok

WDYM

Reply:
meant that we could not get height using HasAt, so I reworked my approach and use Head() directly